tommath.h

```diff
@@ -561,21 +561,21 @@ mp_err mp_prime_next_prime(mp_int *a, int t, bool bbs_style) MP_WUR;
 mp_err mp_prime_rand(mp_int *a, int t, int size, int flags) MP_WUR;
 
 /* Integer logarithm to integer base */
-mp_err mp_log_u32(const mp_int *a, uint32_t base, uint32_t *c) MP_WUR;
+mp_err mp_log(const mp_int *a, int base, int *c) MP_WUR;
 
 /* c = a**b */
-mp_err mp_expt_u32(const mp_int *a, uint32_t b, mp_int *c) MP_WUR;
+mp_err mp_expt(const mp_int *a, int b, mp_int *c) MP_WUR;
 
 /* ---> radix conversion <--- */
 int mp_count_bits(const mp_int *a) MP_WUR;
 
 size_t mp_ubin_size(const mp_int *a) MP_WUR;
-mp_err mp_from_ubin(mp_int *a, const unsigned char *buf, size_t size) MP_WUR;
-mp_err mp_to_ubin(const mp_int *a, unsigned char *buf, size_t maxlen, size_t *written) MP_WUR;
+mp_err mp_from_ubin(mp_int *a, const uint8_t *buf, size_t size) MP_WUR;
```

Comment:
how about using `void*` instead of `uint8_t*` for opaque data?

Reply:
Yes, we could do that. Personally, I am using unsigned char/uint8_t for "binary data". And void is reserved for arbitrary memory, like memory where some struct is allocated. I feel that is a bit distinct from binary data.

Reply:
@sjaeckel are you ok to leave it as uint8_t?

Reply:
> Personally, I am using unsigned char/uint8_t for "binary data". And void is reserved for arbitrary memory

TBH I didn't use `uint8_t*` at all in the last years, `void*` is nowadays the synonym for me (which brings the advantage of no typecasts as required in `mtest_opponent.c` (I'm aware why those casts are required... ;-) )).

> like memory where some struct is allocated

I didn't get that part, what do you mean?

> @sjaeckel are you ok to leave it as uint8_t?

it's okay

Reply:
> I didn't get that part, what do you mean?

I meant that I use 'void*' to point to whatever in memory, while I use 'uint8_t*' for some kind of binary/serialized data or byte strings, which are meant to be read byte wise.Therefore I prefer 'uint8_t*' here instead of the more generic 'void*'.